### ISSUE TITLE
Rule proposal: prefer-unshift

### ISSUE THREAD

**Issue by tjx666** [sindresorhus/eslint-plugin-unicorn#2165]
### Description

I see my newbine team member write following code when I code review:

```ts
 sItems.splice(0, 0, { itemType: 'Ungroup' });
```

### Fail

```js
arr.splice(0, 0, {})
```

### Pass

```js
arr.unshift({})
```

### Additional Info

maybe another proposal to suggest:

```js
arr.splice(0, 1)
```

to:

```
arr.shift()
```

**Comment by fisker** [comment]
I like this idea.

Just a reminder, 

1. the fail case should not limit to 3 arguments, there can be more arguments.
2. the second argument can also be a negative number, though I never seen any one use .
3. the object can also be TypedArray, which should still work the same as Array. (which make the rule name good without `-array`)

**Comment by sindresorhus** [comment]
Accepted
